### DISCUSSION TITLE
Select and reorder items from a list

### DISCUSSION THREAD

**Discussion by A11yship** [neuefische/questions#96] [Web]
Hello,

in my App the user should be able to generate a list of current tasks from the list of all tasks. The list of current tasks doesn't have to contain all tasks. It should also be possible to change the order of the entries. At the moment I'm a bit stuck, what might be a good solution for this case.
Adding a checkbox might help to select the tasks but then I don't know how to change the order.

I look forward to your ideas
Hannah

**Comment by CJAkkuec** [comment]
Hey, please link to your Capstone Project Repository on GitHub. Additionally, if possible: add your current Vercel Deployment.

**Reply by A11yship** [reply]
Here is my [Github Repository](https://github.com/A11yship/capstone-project) and the [Vercel Deployment](http://capstone-project-tau.vercel.app/).

**Comment by CJAkkuec** [comment]
> The list of current tasks doesn't have to contain all tasks.

It might benefit you to have a 'universal source of truth' that contains all tasks, then sort according to your needs/wants.

> It should also be possible to change the order of the entries.

You could add a priority key to your object, such as `priority: 1` and sort from low to high or high to low, for example. Or do you want the order to be completely random?

> At the moment I'm a bit stuck, what might be a good solution for this case.

Always depends on what you want to do.

> Adding a checkbox might help to select the tasks but then I don't know how to change the order.

[The sort() method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) is your friend when it comes to the order of things. You could potentially have the order correspond with a Select-Element that handles the type of sorting you want to do. Not sure what you need the checkbox for, unless you want to be able to sort by current/done?

**Reply by A11yship** [reply]
Thank you for your answer. 
My problem is not so much, how to store the information. It's more about what kind of user input/interface i should use.. 
My first idea was to generate a form with the tasks and a checkbox for every task. The user can select the tasks, he want to have. But with this solution he can't change the order. 
Drag and Drop can be a solution for the ordering problem. But I'm unsure how to select with drag and drop and how accessible it will.

**Reply by mpagels** [reply]
You could have two lists. The first list contains all the information about the tasks. Then you have another list, where you just save unique identifies, like an `id` that matches with the tasks in the first list. And the order of the entries of that second list, is your order. And you could change the order for order display.

Here an example

```js
const tasks = [
{ 
  id: 0,
  task: "wash dishes",
},
{
  id: 1,
  task: "play soccer",
},
{
  id: 2,
  task: "learn coding",
}
}

const currentTasks = [ 2, 0 ]

// the frontend should render "lern coding" and then "wash dishes"
```
You then map over  `currentTasks` and use the `numbers` for finding the tasks, with the equal `id`  in `tasks` and render them. The moment you change the order of the numbers in `currentTasks`, the order is changed in the frontend.

**Reply by A11yship** [reply]
Thank you for the answer, but it doesn't answer my question.
My Problem is what kind of interface/front end I should use to get the information from the user, which is stored in the currentTask array.

**Comment by ahohnsen** [comment]
Maybe the user can select tasks as current tasks with a checkbox/button. By selecting the tasks they could move to a new list with all current tasks. There you could maybe display a select-element next to each task where the user can choose the priority/order of the task (e.g. high number = high priority, do it first and low number = low priority, do it last). 

Another idea could be to have an up & down-button ⬆️ ⬇️  next to each task and the user can move tasks in the list by clicking on these buttons.

If you would like to use drag & drop, you can have a look at this library: https://www.npmjs.com/package/react-beautiful-dnd
